src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -1097,10 +1105,35 @@ def create_or_replace_dynamic_table(
         if not is_sql_select_statement(child.queries[0].sql.lower().strip()):
             raise SnowparkClientExceptionMessages.PLAN_CREATE_DYNAMIC_TABLE_FROM_SELECT_ONLY()
 
+        if create_mode == DynamicTableCreateMode.OVERWRITE:
+            replace = True
+            if_not_exists = False
+        elif create_mode == DynamicTableCreateMode.ERROR_IF_EXISTS:
+            replace = False
+            if_not_exists = False
+        elif create_mode == DynamicTableCreateMode.IGNORE:
+            replace = False
+            if_not_exists = True
+        else:
+            # should never reach here
+            raise ValueError(f"Unknown create mode: {create_mode}")  # pragma: no cover
```

Comment:
Nit: This could be more concise, but your method is probably better if a new mode is ever added.
```suggestion
replace = create_mode == DynamicTableCreateMode.OVERWRITE
if_not_exists == create_mode == DynamicTableCreateMode.IGNORE
```

Reply:
I would prefer to keep it this way. Although it is verbose, I would say it is more readable